src/classes/TaskRepository.cls

```diff
@@ -39,6 +39,16 @@ public without sharing class TaskRepository extends SObjectRepository {
         return (List<Task>)Database.query(query);
     }
 
+    public List<Task> searchInAllFields(String searchTerm) {
+        String query = this
+            .addConditionIsClosed(false)
+            .orderBy(Schema.Task.WhoId)
+            .limitCount(10)
+            .getSearchQuery(searchTerm, SObjectRepository.SearchGroup.ALL_FIELDS);
+
+        return (List<Task>)Search.query(query)[0];
```

Comment:
Dynamic SOSL should not be used as it could expose the application to SOSL injection security vulnerabilities.